packages/terra-clinical-header/src/Header.jsx

```diff
@@ -35,40 +37,142 @@ const defaultProps = {
   isSubheader: false,
 };
 
-const Header = ({ title, startContent, endContent, isSubheader, ...customProps }) => {
-  let startElement;
-  if (startContent) {
-    startElement = <div className={cx('flex-end')}>{startContent}</div>;
+class Header extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleOnResize = this.handleOnResize.bind(this);
+    this.state = {
+      startContentInitialWidth: undefined,
+      endContentInitialWidth: undefined,
+      startContentWidth: undefined,
+      endContentWidth: undefined,
+    };
   }
 
-  let titleElement;
-  if (title) {
-    titleElement = (
-      <div className={cx('title-container')}>
-        <h1 className={cx('title')}>
-          {title}
-        </h1>
-      </div>
-    );
+  componentDidMount() {
+    // grab the initial full display widths of the end containers for reference
+    // when resizing the end containers inside handleOnResize
+    const startNode = ReactDOM.findDOMNode(this.startContainerTarget);
```

Comment:
You shouldn't need to use findDOMNode here. You set `this.startContainerTarget` as the dom node already in the ref callback.

Reply:
updated to not use findDOMNode